internal/settings/providers/dynu/provider.go

```diff
@@ -53,25 +59,30 @@ func New(data json.RawMessage, domain, host string,
 
 func (p *provider) isValid() error {
 	switch {
-	case len(p.username) == 0:
+	case p.username == "":
 		return errors.ErrEmptyUsername
-	case len(p.password) == 0:
+	case p.password == "":
 		return errors.ErrEmptyPassword
 	case p.host == "*":
 		return errors.ErrHostWildcard
+	case p.alias != "" && p.host == "@":
+		return errors.ErrHostWithAlias
 	}
 	return nil
 }
 
 func (p *provider) String() string {
-	return utils.ToString(p.domain, p.host, constants.Dynu, p.ipVersion)
+	return utils.ToString(p.domain, p.Host(), constants.Dynu, p.ipVersion)
 }
 
 func (p *provider) Domain() string {
 	return p.domain
 }
 
 func (p *provider) Host() string {
+	if p.alias != "" {
+		return p.alias + "." + p.host
+	}
```

Comment:
Are you sure this is fine?

Should we return alias.host instead of just alias? 🤔